### PR TITLE
PR for Add label, Edit label and delete label.

### PR DESCRIPTION
**Pull request by manismk**
- Added add label, Edit label and delete label functionality
- Based on the label added the drop-down in notes will change
- [Link](https://deploy-preview-10--key-notes.netlify.app/)

Todo
- Add label functionality to notes

### CODE REVIEW

File: src/components/label/LabelContainer.jsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                placeholder="Add Label Here"
+                onChange={(e) => changeHandler(e)}
+                value={labelData.enteredValue}
+              />
+              <button
+                className="btn icon--btn "
+                disabled={!labelData.enteredValue.length}
+                onClick={() => {
+                  clickHandler();
+                }}
```
Comment (JowelTisso) [inline src/components/label/LabelContainer.jsx:70]: Here prefer passing the function name directly as arrow function has no use for now

File: src/components/label/LabelItem.jsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+          <div>
+            <button className="btn icon--btn" onClick={() => setEditing(true)}>
+              <Edit />
+            </button>
+            <button
+              className="btn icon--btn"
+              onClick={() => {
+                handleLabel(
+                  [...labels.filter((label) => label !== labelData)],
+                  user.uid
```
Comment (JowelTisso) [inline src/components/label/LabelItem.jsx:76]: Here labels array will be modified, Prefer to spread the labels before filtering e.g `[...labels].filter()`

File: src/components/sidebar/Sidebar.jsx [hunk 3]
```diff
@@ -4,34 +4,43 @@ import {
...
+            <p className="sidebar--title">Notes</p>
+          </div>
+        </Link>
+        <div className="sidebar--item" onClick={() => setLabelForm(true)}>
+          <LabelOutlined className="icon icon--md" />
+          <p className="sidebar--title">Add Labels</p>
         </div>
-      </Link>
-    </aside>
+        <Link to="/app/archive">
```
Comment (shraddha-amber) [inline src/components/sidebar/Sidebar.jsx:28]: suggestion: can make the routes as constant variables

**Review by shraddha-amber — APPROVED** [review]
Other than jowel's suggestions, the rest is good!

File: src/components/label-multiselect/LabelMultiSelect.jsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      <div className="selectBox" onClick={() => setDropdown((prev) => !prev)}>
+        <select>
+          <option>Select Label</option>
+        </select>
+        <div className="overSelect"></div>
+      </div>
+      <div
+        className={`checkboxes ${showDropdown ? "open" : "close"}`}
+        onMouseLeave={() => setDropdown(false)}
+      >
```
Comment (JowelTisso) [inline src/components/label-multiselect/LabelMultiSelect.jsx:19]: Good use of on mouse leave

File: src/components/label/LabelContainer.jsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  const { user } = useAuth();
+  const { labels } = useNotes();
+
+  const changeHandler = (e) => {
+    setLabelData((prev) => ({
+      ...prev,
+      enteredValue: e.target.value,
+      error: "",
+    }));
+  };
```
Comment (JowelTisso) [inline src/components/label/LabelContainer.jsx:25]: You can destructure event directly in the params, e.g `({target})=>{}`

**Review by JowelTisso — APPROVED** [review]
Rest Looks Good

**Pull request merged**